lib/util.js

```diff
@@ -599,21 +609,40 @@ function findTypedConstructor(value) {
   }
 }
 
+// Creates a subclass and name
+// the constructor as `${clazz} : null prototype`
+function clazzWithNullPrototype(clazz, name) {
+  class NullPrototype extends clazz {
+    get [Symbol.toStringTag]() {
+      return '';
+    }
+  }
+  Object.defineProperty(NullPrototype.prototype.constructor, 'name',
+                        { value: `[${name}: null prototype]` });
+
+  return NullPrototype;
+}
+
```

Comment:
👆 Why have a `clazzWithNullPrototype` that returns a constructor? It looks like it's used as a fallback for when `Object.getPrototypeOf(value)` is `null`, but most prototypes are objects not class constructors. Would it make sense to make an object with the expected properties instead?

Reply:
No, it has to be the class. The reason is that the hidden symbols and so on have to be available so the entries can be inspected. So if a set has a `null-prototype` we could not use `for...of` and so on but this is essential for the actual inspection. So we restore the prototype and make sure the inspection prints it as `null-prototype`.

Reply:
Is there anyway to reuse these classes that are created? It looks like right now we create new classes for each `null` prototype we run into.

Reply:
Memoization should be possible.

Reply:
May be create null class instances when needed and store it in a  object would be good enough I guess. @BridgeAR

Reply:
This should do I guess:

```
let nullInstances = {};

function clazzWithNullPrototype(clazz, name) {
  if(nullInstances[name])
    return nullInstances[name];

  class NullPrototype extends clazz {
    get [Symbol.toStringTag]() {
      return '';
    }
  }
  Object.defineProperty(NullPrototype.prototype.constructor, 'name',
                        { value: `[${name}: null prototype]` });
  nullInstances[name] = NullPrototype;
  return NullPrototype;
}
```

Reply:
It should be a map instead of an object but otherwise, yes:

```js
let lazyNullPrototypecache;

function clazzWithNullPrototype(clazz, name) {
  if (lazyNullPrototypecache === undefined) {
    lazyNullPrototypecache = new Map();
  } else {
    const cachedClass = lazyNullPrototypecache.get(clazz);
    if (cachedClass !== undefined) {
      return cachedClass;
    }
  }
  ...
  lazyNullPrototypecache.set(clazz, NullPrototype);
  return NullPrototype;
}
```

Reply:
Ok, an object would also work if we use the `name` argument as property. But using objects as keys is only possible with a map.

I actually thought about removing the `name` property but the default clazz names could have been tampered with as well. We could argue that showing that name (`clazz.name`) would be better, even if it's tampered with but I have no strong opinion about that.